src/sdk/namespace_fs.js

```diff
@@ -563,11 +564,13 @@ class NamespaceFS {
 
             let count = 1;
             for (let pos = start; pos < end;) {
+                object_sdk.throw_if_aborted();
 
                 // allocate or reuse buffer
                 const remain_size = Math.max(0, end - pos);
                 const { buffer, callback } = await buffers_pool.get_buffer();
```

Comment:
```suggestion
                // TODO buffers_pool and the underlying semaphore should support abort signal
                // to avoid sleeping inside the semaphore until the timeout while the request is already aborted.
                const { buffer, callback } = await buffers_pool.get_buffer();
```